src/Sylius/Bundle/ApiBundle/composer.json

```diff
@@ -24,11 +24,12 @@
     ],
     "require": {
         "php": "^8.0",
-        "doctrine/dbal": "^2.7 || ^3.0",
         "api-platform/core": "^2.6",
+        "doctrine/dbal": "^2.7 || ^3.0",
         "enshrined/svg-sanitize": "^0.15.4",
         "lexik/jwt-authentication-bundle": "^2.11",
         "sylius/core-bundle": "^1.12",
+        "symfony/doctrine-messenger": "^5.4 || ^6.0",
```

Comment:
Why? If someone uses a different transport this will be a dead dependency.

Reply:
I thought if a dependency was added and the release occurred, we couldn't remove it as it became a hard dependency. (I assumed that someone's using Sylius 1.12 with doctrine transport onboard)
I also added the dependency to `ApiBundle` since it's already present in the root `composer.json`

Reply:
That's kinda beside the point since if you're using the full Sylius you'll have it either way, but if you were using only ApiBundle, now you'll have an extra dependency.
Also why only in Api when the config/migration utilizing it is in Core?
Personally, I'd be for removing it from the root composer in 1.13.